sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1219,7 +1241,8 @@ class Analyzer(
 
       // Make sure a plan's expressions do not contain outer references
       def failOnOuterReference(p: LogicalPlan): Unit = {
-        if (p.expressions.exists(containsOuter)) {
+        p.expressions.foreach(checkMixedReferencesInsideAggregation)
```

Comment:
@viirya checkMixedReferencesInsideAggregation only looks for AggregateExpression ? For other types its just a pass-through.